core/src/main/java/org/apache/calcite/rel/type/RelDataTypeSystem.java

```diff
@@ -132,11 +132,19 @@ RelDataType deriveCovarType(RelDataTypeFactory typeFactory,
    * @return the result type for a decimal addition.
    */
   default RelDataType deriveDecimalPlusType(RelDataTypeFactory typeFactory,
-                                            RelDataType type1, RelDataType type2) {
+      RelDataType type1, RelDataType type2) {
     if (SqlTypeUtil.isExactNumeric(type1)
             && SqlTypeUtil.isExactNumeric(type2)) {
       if (SqlTypeUtil.isDecimal(type1)
               || SqlTypeUtil.isDecimal(type2)) {
+        // Java numeric will always have invalid precision/scale,
+        // use its default decimal precision/scale instead.
+        type1 = RelDataTypeFactoryImpl.isJavaType(type1)
```

Comment:
Hi, @danny0405. Maybe we also need to handle `deriveDecimalMultiplyType` and `deriveDecimalDivideType` in this class. I wait for your fix here to complete CALCITE-3245.

Reply:
Sure, let me add them in, i believe this patch would be merged soon.